test/transactions/transfer.js

```diff
@@ -16,42 +15,168 @@ describe('transfer.js', function () {
+				.with.property('dappId')
+				.equal(dappId);
+		});
+
+		it('should create an in transfer dapp transaction with a provided amount', function () {
+			(inTransferTransaction).should.have.property('amount').equal(amount);
+		});
+
+		it('should create an in transfer dapp transaction with a default fee', function () {
+			(inTransferTransaction).should.have.property('fee').equal(0.1e8);
+		});
+
+		it('should create an in transfer dapp transaction with no recipient', function () {
+			(inTransferTransaction).should.have.property('recipientId').be.null();
+		});
+
+		it('should create an in transfer dapp transaction with senderPublicKey', function () {
+			(inTransferTransaction).should.have.property('senderPublicKey').equal(publicKey);
+		});
+
+		it('should create an in transfer dapp transaction with first signature', function () {
+			(inTransferTransaction).should.have.property('signature').and.be.ok();
+		});
+
+		it('should create an in transfer dapp transaction with second signature', function () {
+			(inTransferTransaction).should.have.property('signSignature').and.be.ok();
+		});
+
+		it('should create an in transfer dapp transaction with just one signature', function () {
+			var inTransferTransactionOneSignature = createInTransfer(dappId, amount, secret);
+			(inTransferTransactionOneSignature).should.have.property('signature').and.be.ok();
+			(inTransferTransactionOneSignature).should.not.have.property('secondSignature');
+		});
+
+		it('should use time slots to get the time for the timestamp', function () {
+			var now = new Date();
+			var clock = sinon.useFakeTimers(now, 'Date');
+			var time = 36174862;
+			var stub = sinon.stub(slots, 'getTime').returns(time);
+
+			var trs = createInTransfer(dappId, amount, secret);
+
+			(trs).should.have.property('timestamp').and.be.equal(time);
+			(stub.calledWithExactly(now.getTime())).should.be.true();
+
+			stub.restore();
+			clock.restore();
+		});
+
+		it('should use time slots with an offset to get the time for the timestamp', function () {
+			var now = new Date();
+			var clock = sinon.useFakeTimers(now, 'Date');
+			var offset = 10e3;
+			var time = 36174862;
+			var stub = sinon.stub(slots, 'getTime').returns(time);
+
+			var trs = createInTransfer(dappId, amount, secret, null, offset);
+
+			(trs).should.have.property('timestamp').and.be.equal(time);
+			(stub.calledWithExactly(now.getTime() - offset)).should.be.true();
+
+			stub.restore();
+			clock.restore();
+		});
+
+	});
+
+	describe('#createOutTransfer', function () {
+		var createOutTransfer = transfer.createOutTransfer;
+		var dappId = '1234213';
+		var transactionId = '9876567';
+		var recipientId = '989234L';
+		var amount = 10e8;
+		var secret = 'secret';
+		var publicKey = '5d036a858ce89f844491762eb89e2bfbd50a4a0a0da658e4b2628b25b117ae09';
+		var secondSecret = 'secondSecret';
+		var outTransferTransaction = createOutTransfer(dappId, transactionId, recipientId, amount, secret, secondSecret);
```

Comment:
These tests will be restructured as part of https://github.com/LiskHQ/lisk-js/issues/178 - are you happy if we do that then rather than now? (I agree it would be nice.)